CRM/MembershipExtras/Job/OfflineAutoRenewal/MultipleInstallmentPlan.php

```diff
@@ -46,7 +46,7 @@ protected function getRecurringContributions() {
           ccr.contribution_status_id != {$cancelledStatusID} 
           AND ccr.contribution_status_id != {$refundedStatusID}
          )
-         AND ppp.next_period IS NULL
+         AND (ppp.next_period IS NULL OR ppp.next_period = 0)
```

Comment:
should SingInstallmentPlan be updated too, Also I think it is better if we update the upgrader to set the value to 'Null' too at the same time.

Reply:
Single installment plans only have one recurring contribution. Each time it gets renewed, a new contribution is added to the same recurring contribution. Thus for single installment payment plans, there is no previous or next periods (these are actually references to the prior recurring contribution and the next recurring contribution).